src/components/Garden/ModalFlows/ModalFlowBase.tsx

```diff
@@ -135,17 +143,16 @@ function ModalFlowBase({
   return <MultiModalScreens screens={extendedScreens} />
 }
 
-/* eslint-disable react/prop-types */
-function LoadingScreen({ loading }) {
+function LoadingScreen({ loading }: { loading: boolean }) {
   const theme = useTheme()
   const { next } = useMultiModal()
 
   useEffect(() => {
-    let timeout
+    let timeout: number
 
     if (!loading) {
       // Provide a minimum appearance duration to avoid visual confusion on very fast requests
-      timeout = setTimeout(() => {
+      timeout = window.setTimeout(() => {
```

Comment:
Because we migrate this file to tsx, and since timeout is a number, we cannot assign a timeout variable to a setTimeout. The ts error is the following. `Type 'Timeout' is not assignable to type 'number' `

Reply:
`Nodejs.setTimeout()
`
![image](https://user-images.githubusercontent.com/691510/148255584-e6fa2af1-56f5-420e-a30b-08f3bffca794.png)

Exist 2 types of setTimeout, one from NodeJS and another from window, browser side.
So the types changes a bit.

`window.setTimeout()
`
![image](https://user-images.githubusercontent.com/691510/148255743-a923ead3-5c36-44a3-98b6-26d96001c9ea.png)

Reply:
The explanation of Felipe is correct, although in this case we use only client side, and the return is a positive integer value, that can be used by clearTimeout() to cancel the timer.